core/src/agent/actions/commit.rs

```diff
@@ -0,0 +1,60 @@
+extern crate futures;
+use action::{Action, ActionWrapper};
+use agent::state::ActionResponse;
+use context::Context;
+use futures::Future;
+use hash_table::entry::Entry;
+use instance::dispatch_action;
+use std::sync::{mpsc::Sender, Arc};
+
+/// Commit Action Creator
+/// This is the high-level commit function that wraps the whole commit process and is what should
+/// be called from zome api functions and other contexts that don't care about implementation details.
+///
+/// Returns a future that resolves to an ActionResponse.
+pub fn commit_entry(
```

Comment:
Maybe prefix the fn name with `dispatch_` or some other convention to make clear its not actually doing the work directly.
We already use `reduce_` and `invoke_` for the other Commit related functions.

Reply:
hm, but when used with future composers it would read strange:
```
block_on(
  dispatch_validate_entry()
    .and_then(|_|{ 
      dispatch_commit_entry() 
    }
  );
```
because it does not continue after the dispatch, but after the whole task was done.

Reply:
How about `async_` or `await_` or `promise_` or `future_` or ...

I think my preference is `promise_`

Reply:
@ddd-mtl the thing is that `async` `await` `promise` `future` are all Coming Soon in rust core :/

Reply:
what if this was more like `CommitFuture::from_entry()`?

Reply:
@thedavidmeister I know, that's exactly why I proposed that those prefixs. But I definitely prefer your idea. I'll add that its also with Context, so maybe more like `CommitFuture::from_entry_with_context()`

Reply:
Well.. I don't like focusing on the Future aspect so much that we represent that in the name of these functions.

After our Redux conversations, I actually think we should go the whole way to pure reducers which makes total sense since we now have these functions as *Action Creators*. Any asynchronous running of zome functions etc. should actually happen in here. So `validate_entry()` dispatches a `ValidationStarted` action **and** kicks-off the worker thread for the validation (which right now is in the reducer and makes it be not a pure function). That worker thread also dispatches a `ValidationResult` action when it's done. Reducers only take those action parameters and update the state.

I would leave the name for the action creators here as they are now, since they are scoped in action already, i.e. `nucleus::actions::validate_entry(entry.clone()).and_then(....`.